### PR TITLE
NFSD Outpost Criminal Records Fix

### PR DESCRIPTION
**Pull request by Qulibly**
## About the PR
A split from https://github.com/new-frontiers-14/frontier-station-14/pull/1578

It allows proper function of criminal records by adding a workaround which lets NFSD Outpost receive a record of each player that spawned in no matter in what location, can be Frontier Outpost, Latejoining vessel etc.

This way NFSD Outpost has records of every single player, and criminal record console it is possible to write notes on player records as well as set their wanted level which works globally(from testing)

Adds system for Pirates and future antags. Where it can either fake the records, AKA real job won't be visible and replaced with one of the three most common jobs (contractor, pilot, mercenary), OR where they won't be registered in record at all.

## Why / Balance

NFSD greatly lacks managing as well as intel gathering tools. 
This will mainly assist detectives in forensics. Sherrifs in writing more accurate warrants or marking wanted/released folk.

But mostly assist with archiving infractions done by each person.

## How to test

Join any station/ship crew.
Go to NFSD Outpost
Check in station records computer if person was added
spawn criminal record console on NFSD
check if player information shows up
check if changing status has some effect

## Requirements

- [X] I have read and am following the [Pull Request and Changelog Guidelines](https://docs.spacestation14.com/en/general-development/codebase-info/pull-request-guidelines.html).
- [X] I have added media to this PR or it does not require an ingame showcase.

## Breaking changes

Anything that affects Station Records or Criminal Records.

**Changelog**

:cl:
- tweak: Criminal records computers now work sector-wide, and contain information for all employees.

### CODE REVIEW

**Comment by Qulibly** [comment]
Additionally for this a criminal record console may need to be mapped onto NFSD Outpost

File: Resources/Prototypes/_NF/SectorServices/services.yml [hunk 1]
```diff
@@ -1,28 +1,14 @@
 - type: sectorService
   id: PirateBounties
   components:
-  - type: SectorPirateBountyDatabase
+    - type: SectorPirateBountyDatabase
```
Comment (dvir001) [inline Resources/Prototypes/_NF/SectorServices/services.yml:4]: No extra spaces please

File: Resources/Prototypes/_NF/SectorServices/services.yml [hunk 2]
```diff
@@ -1,28 +1,14 @@
...
 
 - type: sectorService
   id: MailMetrics
   components:
-  - type: SectorLogisticStats
+    - type: SectorLogisticStats
```
Comment (dvir001) [inline Resources/Prototypes/_NF/SectorServices/services.yml:9]: No extra spaces please

File: Content.Server/StationRecords/Systems/StationRecordsSystem.cs [hunk 3]
```diff
@@ -11,6 +12,7 @@
 using Content.Shared.StationRecords;
 using Robust.Shared.Enums;
 using Robust.Shared.Prototypes;
+using Robust.Shared.Random; //Frontier modification
```
Comment (dvir001) [inline Content.Server/StationRecords/Systems/StationRecordsSystem.cs:15]: ``// Frontier``

File: Content.Server/StationRecords/Systems/StationRecordsSystem.cs [hunk 4]
```diff
@@ -54,6 +57,28 @@ private void OnPlayerSpawn(PlayerSpawnCompleteEvent args)
...
+                if (TryComp<StationRecordKeyStorageComponent>(targetId, out var keyStorage)
+                && stationEntityUid != null
+                && keyStorage.Key != null)
+                {
+                    if (!TryGetRecord<GeneralStationRecord>(Key.Value, out var record))
+                        continue;
+            }
+            }
+
+        }*/
```
Comment (dvir001) [inline Content.Server/StationRecords/Systems/StationRecordsSystem.cs:81]: ?

File: Content.Server/StationRecords/Systems/StationRecordsSystem.cs [hunk 5]
```diff
@@ -95,6 +120,44 @@ private void CreateGeneralRecord(EntityUid station, EntityUid player, HumanoidCh
...
+                {
+                    string[] jobs = { "Contractor", "Pilot", "Mercenary" };
+                    // Randomises job
+                    var random = _robustRandom.Next(jobs.Length);
+                    playerJob = jobs[random];
+                }
+                CreateGeneralRecord(stationEntityUid, idUid.Value, profile.Name, profile.Age, profile.Species, profile.Gender, playerJob, fingerprintComponent?.Fingerprint, dnaComponent?.DNA, profile, stationRec!);
+            }
+
+        }
```
Comment (dvir001) [inline Content.Server/StationRecords/Systems/StationRecordsSystem.cs:160]: ``// Frontier`` block of code

**Review by dvir001 — DISMISSED** [review]

File: Content.Server/StationRecords/Systems/StationRecordsSystem.cs (hunk 4 above)
Comment (Qulibly) [reply Content.Server/StationRecords/Systems/StationRecordsSystem.cs:81]: With this it was original block present. 
And existed as commented to assist with upstream merges and merge conflicts that could happen.

As changed decently below.
Though if this part doesn't need to stay then can remove it

**Comment by Qulibly** [comment]
Okay should have fixed everything that was requested. 
Will test it soonish to ensure that it still runs properly

**Comment by whatston3** [comment]
I'll look into the test failure.

**Comment by Qulibly** [comment]
> I'll look into the test failure.

Got it
So far have pulled it and ran.
It is working great.

The criminal records are accessable on outpost AND other vessels as well. Which is amazing!

Sechud also works flawlessly as well as the crim history.

So if there arent more issues could perhaps merge master and check then again?

File: Content.Server/StationRecords/Systems/StationRecordsSystem.cs (hunk 5 above)
Comment (Qulibly) [reply Content.Server/StationRecords/Systems/StationRecordsSystem.cs:126]: The changes that were added and then by Whatston3 should mark it correctly

**Comment by whatston3** [comment]
"haha we create a small entity for the services"
doesn't destroy it
_doesn't destroy it_

oops

**Review by whatston3 — APPROVED** [review]

**Comment by whatston3** [comment]
Retested, looks fine.  Merging.

**Pull request merged**